app/models/spree/slide.rb

```diff
@@ -29,6 +34,21 @@ def slide_link
   end
 
   def slide_image
-    !image.file? && product.present? && product.images.any? ? product.images.first.attachment : image
+    !image.attached? && product.present? && product.images.any? ? product.images.first.attachment : image.attachment
+  end
+
+  # Helper for resizing
+  def preview
+    image_variant(:preview)
+  end
+
+  def thumbnail
+    image_variant(:thumbnail)
+  end
+
+  private
+
+  def image_variant(variant)
```

Comment:
could we change this method and param name ? in Spree we have `Variant` model and this method isn't related